packages/swirl-components/src/components/swirl-popover-trigger/swirl-popover-trigger.tsx

```diff
@@ -145,7 +159,34 @@ export class SwirlPopoverTrigger {
     return isActive;
   }
 
+  private getPopoverAndTriggerEls() {
+    const popoverEl = this.getPopoverEl();
+
+    if (!Boolean(popoverEl)) {
+      return { popoverEl: null, triggerEl: null };
+    }
+
+    const triggerEl = this.getTriggerEl();
+
+    if (!Boolean(triggerEl)) {
+      return { popoverEl: null, triggerEl: null };
+    }
+
+    return { popoverEl, triggerEl };
+  }
+
   render() {
+    if (this.triggerOnHover) {
```

Comment:
Suggestion: Skip the if and just do:

```
<Host
  onClick={this.onClick}
  onMouseenter={this.onMouseenter}
  onMouseleave={this.onMouseleave }
>
  <slot></slot>
</Host>
```
➕ add an early return in onMouseenter and onMouseleave if triggerOnHover is false.